components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -456,4 +481,39 @@ private boolean isValidEndpointDefinition(URL definitionURL, InputStream definit
         return isValid;
     }
 
+    @Override
+    public Response getEndpointDefinition(String registryId, String entryId, MessageContext messageContext) {
+        String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        String contentType = StringUtils.EMPTY;
+        EndpointRegistry registryProvider = new EndpointRegistryImpl();
+        InputStream endpointDefinition = null;
+        try {
+            EndpointRegistryInfo endpointRegistry = registryProvider.getEndpointRegistryByUUID(registryId, tenantDomain);
+            if (endpointRegistry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry with the id: " + registryId +
+                        " is not found", log);
+            }
+            EndpointRegistryEntry registryEntry = registryProvider.getEndpointRegistryEntryByUUID(registryId, entryId);
+            if (registryEntry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry entry with the id: " + entryId +
+                        " is not found", log);
+            }
+            String type = registryEntry.getDefinitionType();
+            if (RegistryEntryDTO.DefinitionTypeEnum.OAS.equals(RegistryEntryDTO.DefinitionTypeEnum.fromValue(type))
+                    || RegistryEntryDTO.DefinitionTypeEnum.GQL_SDL.equals(RegistryEntryDTO.DefinitionTypeEnum
+                    .fromValue(type))) {
+                contentType = MediaType.APPLICATION_JSON;
+            } else if (RegistryEntryDTO.DefinitionTypeEnum.WSDL1.equals(RegistryEntryDTO.DefinitionTypeEnum
+                    .fromValue(type)) || RegistryEntryDTO.DefinitionTypeEnum.WSDL2.equals(RegistryEntryDTO
+                    .DefinitionTypeEnum.fromValue(type))) {
+                contentType = MediaType.TEXT_XML;
+            }
+            endpointDefinition = registryEntry.getEndpointDefinition();
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while retrieving the endpoint definition of registry " +
+                    "entry with id: " + entryId, e, log);
+        }
+        return Response.ok(endpointDefinition).type(contentType).build();
```

Comment:
when endpointDefinition == null, need to throw an error response right?

Reply:
Fixed in 247029c